File: framework/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/model/ECommerceProductReference.java
package com.sdl.ecommerce.dxa.model;

import com.sdl.ecommerce.api.model.ECommerceEclItem;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * ECommerce Product Refererence
 *
 * @author nic
 */
@SemanticEntity(entityName = "ECommerceProduct", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class ECommerceProductReference extends AbstractEntityModel {

    @SemanticProperty("e:productId")
    private String productId;

    @SemanticProperty("e:productRef")
    private ECommerceEclItem productRef;

    public String getProductId() {
        return productId;
    }

    public ECommerceEclItem getProductRef() {
        return productRef;
    }

}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/ECommerceProductReference.cs
using Sdl.Web.Common.Models;

namespace SDL.ECommerce.DXA.Models
{
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "ECommerceProduct", Prefix = "e")]
    public class ECommerceProductReference : EntityModel
    {
        [SemanticProperty("e:productId")]
        public string ProductId { get; set; }

        [SemanticProperty("e:productRef")]
        public ECommerceEclItem ProductRef { get; set; }
    }
}
